cmd/meroxa/turbine/golang/deploy.go

```diff
@@ -2,144 +2,122 @@ package turbinego
+		})
 	}
-
 	return resources, nil
 }
 
 // NeedsToBuild reads from the Turbine application to determine whether it needs to be built or not
 // this is currently based on the number of functions.
-func (t *turbineGoCLI) NeedsToBuild(ctx context.Context, appName string) (bool, error) {
-	cmd := exec.CommandContext(ctx, path.Join(t.appPath, appName), "--listfunctions") //nolint:gosec
-
-	accessToken, refreshToken, err := global.GetUserToken()
+// TODO: Once all languages are under turbine-core refactor this so it's the same for all languages.
+func (t *turbineGoCLI) NeedsToBuild(ctx context.Context) (bool, error) {
+	ctx, cancel := context.WithTimeout(ctx, time.Second)
+	defer cancel()
+	resp, err := t.bc.HasFunctions(ctx, &emptypb.Empty{})
 	if err != nil {
 		return false, err
 	}
-	cmd.Env = os.Environ()
-	cmd.Env = append(cmd.Env, fmt.Sprintf("ACCESS_TOKEN=%s", accessToken), fmt.Sprintf("REFRESH_TOKEN=%s", refreshToken))
 
-	output, err := cmd.CombinedOutput()
-	if err != nil {
-		fmt.Println(string(output))
-		return false, fmt.Errorf("build failed")
-	}
-	list, err := utils.GetTurbineResponseFromOutput(string(output))
-	if err == nil && list != "" {
-		// ignores any lines that are not intended to be part of the response
-		list = strings.Trim(list, "[]")
-		if list == "" {
-			return false, nil
-		}
-
-		hasFunctions := len(strings.Split(list, ",")) > 0
-		return hasFunctions, nil
-	}
-
-	re := regexp.MustCompile(`\[(.+?)]`)
-	// stdout is expected as `"2022/03/14 17:33:06 available functions: []` where within [], there will be each function.
-	hasFunctions := len(re.FindAllString(string(output), -1)) > 0
-
-	return hasFunctions, nil
+	return resp.Value, nil
 }
 
 func (t *turbineGoCLI) GetGitSha(ctx context.Context) (string, error) {
-	return utils.GetGitSha(ctx, t.appPath)
+	return turbine.GetGitSha(ctx, t.appPath)
 }
 
 func (t *turbineGoCLI) GitChecks(ctx context.Context) error {
-	return utils.GitChecks(ctx, t.logger, t.appPath)
+	return turbine.GitChecks(ctx, t.logger, t.appPath)
 }
 
-func (t *turbineGoCLI) CreateDockerfile(_ context.Context, appName string) (string, error) {
-	return t.appPath, turbineGo.CreateDockerfile(appName, t.appPath)
+func (t *turbineGoCLI) CreateDockerfile(_ context.Context, appName, specVersion string) (string, error) {
+	return t.appPath, turbineGo.CreateDockerfile(appName, t.appPath, specVersion)
 }
 
 func (t *turbineGoCLI) CleanUpBuild(_ context.Context) {
-	utils.CleanupDockerfile(t.logger, t.appPath)
+	turbine.CleanupDockerfile(t.logger, t.appPath)
 }
 
-func (t *turbineGoCLI) SetupForDeploy(_ context.Context, _ string) (func(), error) {
-	return func() {}, nil
+func (t *turbineGoCLI) SetupForDeploy(ctx context.Context, gitSha string) (func(), error) {
+	source := rand.NewSource(time.Now().UnixNano())
+	rand := rand.New(source)
+	port := rand.Intn(65535-1024) + 1024 //nolint:gomnd
+	grpcListenAddress := fmt.Sprintf("localhost:%d", port)
+
+	go t.builder.RunWithAddress(ctx, grpcListenAddress)
```

Comment:
the grpc address can be set when the builder is instantiated

Reply:
yes, you're right

Reply:
@lyuboxa https://github.com/meroxa/cli/pull/708/commits/3502dd6a44da206109cf01d6869ced7e23f29fa9